scripts/gen_operator_csv.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python
+#
+# Generate an operator bundle for publishing to OLM. Copies appropriate files
+# into a directory, and composes the ClusterServiceVersion which needs bits and
+# pieces of our rbac and deployment files.
+#
+# Usage ./scripts/gen_operator_csv.py OUTPUT_DIR PREVIOUS_VERSION IMAGE_NAME
+# Example: scripts/gen_operator_csv.py tmp 0.1 quay.io/redhat/dedicated-admin-operator:latest
+
+import datetime
+import os
+import sys
+import yaml
+import shutil
+import subprocess
+
+
+def get_git_sha():
+    sha = subprocess.check_output('git rev-parse HEAD', shell=True)
+    return str(sha)[0:7]
+
+def get_num_commits():
+    num = subprocess.check_output('git rev-list HEAD --count', shell=True)
+    return num.rstrip()
+
+
+if __name__ == '__main__':
+
+    # This script will append the current number of commits given as an arg
+    # (presumably since some past base tag), and the git hash arg for a final
+    # version like: 0.1.189-3f73a592
+    VERSION_BASE = "0.1"
+
+    if len(sys.argv) != 4:
+        print("USAGE: %s OUTPUT_DIR PREVIOUS_VERSION IMAGE_NAME" % sys.argv[0])
+        sys.exit(1)
+
+    outdir = sys.argv[1]
+    prev_version = sys.argv[2]
+    operator_image = sys.argv[3]
+    git_num_commits = get_num_commits()
+    git_sha = get_git_sha()
+
+    full_version = "%s.%s-%s" % (VERSION_BASE, git_num_commits, git_sha)
+    print("Generating CSV for version: %s" % full_version)
+
+    if not os.path.exists(outdir):
+        os.mkdir(outdir)
+
+    version_dir = os.path.join(outdir, full_version)
+    if not os.path.exists(version_dir):
+        os.mkdir(version_dir)
+
+    with open('scripts/templates/csv-template.yaml', 'r') as stream:
+        csv = yaml.load(stream)
+
+    csv['spec']['install']['spec']['clusterPermissions'] = []
+
+    cluster_roles = [
+        {'permissions': 'manifests/05-dedicated-admin-operator.ClusterRole.yaml', 'serviceAccountName': 'dedicated-admin-operator'}
```

Comment:
A little concerned we have the filenames hard coded.  Rather we look at all yaml files in manifests and figure out which to pull in based on kind.  If we need to change the order these filenames are likely to change, making this script brittle.

Reply:
Agreed with Naveen on this one.

Reply:
removed hardcoded references to files, now we loop thru the files and based on kind and metadata.name we pick the right objects.